src/main/java/filter/expression/Qualifier.java

```diff
@@ -398,6 +399,15 @@ private static boolean isMetaQualifier(Qualifier q) {
             case "date":
                 comparator = (a, b) -> a.getUpdatedAt().compareTo(b.getUpdatedAt());
                 break;
+            case "nonselfupdate":
+                // Purposefully inverted a and b to make latest issues show up first.
+                if (metadataRefresh) {
+                    comparator = (a, b) ->
+                        b.getMetadata().getNonSelfUpdatedAt().compareTo(a.getMetadata().getNonSelfUpdatedAt());
+                } else {
+                    comparator = (a, b) -> b.getUpdatedAt().compareTo(a.getUpdatedAt());
+                }
+                break;
```

Comment:
Adding a case here has the side effect of making `sort:nonselfupdate` valid syntax, which is probably not what we want.

Since there is only one instance of "nonselfupdate" as a key and `metadataRefresh` is constant there, maybe you could create another method to handle it? `getUpdateComparator`?

Also, since the comparator brought on by `updated` can never compose with another `sort` comparator, there's no need to modify `getCompoundSortComparator`.